app/src/main/java/org/schabi/newpipe/util/JSONSerializer.java

```diff
@@ -0,0 +1,37 @@
+package org.schabi.newpipe.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+
+/**
+ * Centralized class for JSON serialization.
+ */
+public final class JSONSerializer {
+    private JSONSerializer() {
+
+    }
+
+    private static ObjectMapper newMapper() {
+        return new ObjectMapper();
+    }
+
+    public static void toJson(final Object objToConvert, final OutputStream target) {
+        try {
+            newMapper().writeValue(target, objToConvert);
+        } catch (final IOException ioe) {
+            throw new UncheckedIOException(ioe);
+        }
```

Comment:
Any reason to wrap it in `UncheckedIOException`?

It seems that exception is used in the stream API and ideally one should also throw it there.
https://stackoverflow.com/a/23402640